PowerControl/Controller.cs

```diff
@@ -142,6 +169,23 @@ public Controller()
                 dismissNeptuneInput();
             });
 
+            GlobalHotKey.RegisterHotKey(Settings.Default.MenuToggle, () =>
+            {
+                isOSDToggled = !isOSDToggled;
```

Comment:
This seems that it should rather work on top of `rootMenu.Visible`.

```csharp
GlobalHotKey.RegisterHotKey(Settings.Default.MenuToggle, () =>
{
    if (!RTSS.IsOSDForeground())
        return;

    if (rootMenu.Visible)
        hideOSD();
    else
        showOSD();
});
```

Reply:
But then either toggle won't be actual toggle, but rather a show with a very long dismiss time, or the whole OSD won't be dismissable. The isOSDToggled is used in hideOSD method

Reply:
Hmm. It is a little messy - the current implementation. So. I guess we should still do: `isOSDToggled = !rootMenu.Visible`.